iroh-net/src/magicsock/relay_actor.rs

```diff
@@ -316,6 +318,7 @@ impl RelayActor {
                     trace!("tick: cleanup");
                     with_cancel(self.cancel_token.child_token(), self.clean_stale_relay()).await;
                 }
+                // This case will never hit. TODO: Figure out what the intention was here
```

Comment:
deal with `join_next` returning `None` or `receiver.recv()` returning `None`

Reply:
I left a comment on the `join_next` case - it returning `None` isn't actually a special case so we don't really need to handle that differently